x-pack/plugins/security_solution/public/resolver/view/panels/event_detail.tsx

```diff
@@ -42,28 +42,29 @@ export const EventDetail = memo(function EventDetail({
   /** The event type to show in the breadcrumbs */
   eventType: string;
 }) {
-  const event = useSelector((state: ResolverState) =>
-    selectors.eventByID(state)({ nodeID, eventID })
-  );
+  const isEventLoading = useSelector(selectors.isCurrentRelatedEventLoading);
+  const isProcessTreeLoading = useSelector(selectors.isTreeLoading);
+
+  const isLoading = isEventLoading || isProcessTreeLoading;
+
+  const event = useSelector(selectors.currentRelatedEventData);
   const processEvent = useSelector((state: ResolverState) =>
     selectors.processEventForID(state)(nodeID)
   );
-  if (event && processEvent) {
-    return (
-      <EventDetailContents
-        nodeID={nodeID}
-        event={event}
-        processEvent={processEvent}
-        eventType={eventType}
-      />
-    );
-  } else {
-    return (
-      <StyledPanel>
-        <PanelLoading />
-      </StyledPanel>
-    );
-  }
+  return isLoading ? (
+    <StyledPanel>
+      <PanelLoading />
+    </StyledPanel>
+  ) : event ? (
+    <EventDetailContents
```

Comment:
I think you need `StyledPanel` here as well.

Reply:
It's the top level wrapper within the `EventDetailContents` component